scene/src/components/backpack/WearableCatalogGrid.tsx

```diff
@@ -0,0 +1,317 @@
+                  positionType: 'absolute',
+                  position: {
+                    top: -16 * canvasScaleRatio,
+                    left: -12 * canvasScaleRatio
+                  },
+                  padding: {
+                    top: 18 * canvasScaleRatio,
+                    left: 13 * canvasScaleRatio
+                  },
+                  zIndex: 2,
+                  flexDirection: 'column',
+                  pointerFilter: 'none'
+                }}
+                uiBackground={SELECTED_BACKGROUND}
+              >
+                <WearableCellThumbnail
+                  uiTransform={{
+                    width: canvasScaleRatio * 210,
+                    height: canvasScaleRatio * 210
+                  }}
+                  catalystWearable={wearableElement}
+                  canvasScaleRatio={canvasScaleRatio}
+                  loading={loading}
+                />
+                {isEquipped(wearableElement, equippedWearables, baseBody) &&
+                wearableElement.category ===
+                  WEARABLE_CATEGORY_DEFINITIONS.body_shape.id ? null : (
+                  <RoundedButton
+                    uiTransform={{
+                      margin: {
+                        top: 9 * canvasScaleRatio,
+                        left: 1
+                      },
+                      width: 206 * canvasScaleRatio,
+                      height: 60 * canvasScaleRatio
+                    }}
+                    fontSize={26 * canvasScaleRatio}
+                    text={
+                      isEquipped(wearableElement, equippedWearables, baseBody)
+                        ? 'UNEQUIP'
+                        : 'EQUIP'
+                    }
+                    isSecondary={isEquipped(
+                      wearableElement,
+                      equippedWearables,
+                      baseBody
+                    )}
+                    onClick={() => {
+                      if (
+                        isEquipped(wearableElement, equippedWearables, baseBody)
+                      ) {
+                        void onUnequipWearable(wearableElement)
+                      } else {
+                        void onEquipWearable(wearableElement)
+                      }
+                    }}
+                  />
+                )}
+              </UiEntity>
+            ) : null}
+          </UiEntity>
+        )
+      })}
+    </UiEntity>
+  )
+}
+
+function isSelected(wearableURN: URNWithoutTokenId): boolean {
+  return state.selectedWearableURN === wearableURN
+}
+
+function select(wearableURNWithoutTokenId: null | URNWithoutTokenId): void {
+  state.selectedWearableURN = wearableURNWithoutTokenId
+}
+
+const isEquippedMemo: {
+  equippedWearables: URNWithoutTokenId[]
+  baseBody: PBAvatarBase
+  memo: Record<URNWithoutTokenId, boolean> // TODO consider using Map if possible for performance improvement because long keys
+} = {
+  equippedWearables: [],
+  memo: {},
```

Comment:
```suggestion
  memo: new Map<URNWithoutTokenId, boolean>(),,
```

would be great to use a Map here, and use the functions `has(), get(), includes(), set()`, and also it is well-defined

Reply:
https://github.com/dclexplorer/bevy-ui-scene/commit/f5dcdfe02e7ac3ab1f17985e7e20fcab4cd14407